core/unit_test/CMakeLists.txt

```diff
@@ -13,9 +13,14 @@ KOKKOS_ADD_TEST_LIBRARY(
   SOURCES ${GTEST_SOURCE_DIR}/gtest/gtest-all.cc
 )
 
-# WORKAROUND FOR HIPCC
 IF(Kokkos_ENABLE_HIP)
-  TARGET_COMPILE_DEFINITIONS(kokkos_gtest PUBLIC "-DGTEST_HAS_PTHREAD=0 --amdgpu-target=gfx906")
+  IF("${KOKKOS_CXX_COMPILER_ID}" STREQUAL Clang)
+    TARGET_COMPILE_DEFINITIONS(kokkos_gtest PUBLIC "-DGTEST_HAS_PTHREAD=0")
+    SET_SOURCE_FILES_PROPERTIES(${GTEST_SOURCE_DIR}/gtest/gtest-all.cc COMPILE_FLAGS "--cuda-host-only")
```

Comment:
Why is this different from how you do it in `algorithms/unit_tests/CMakeLists.txt`?

Reply:
Good catch, I think I forgot to update it to match.

Reply:
Fixed in 7c58c9b

Reply:
Turns out this was masking that I wasn't detecting HCC vs HIP-Clang correctly -- the so called "correct" way to do it is to look at the value of `HIP_COMPILER` after running find_package(HIP) -- however, this made me discover that `--cuda-host-only` is valid in HCC as well.

Reply:
... so, come to find out that there was a method to my madness here after all.  `SET_SOURCE_FILES_PROPERTIES` is necessary such that the `--cuda-host-only` flag is not applied to _all_ the HIP unit test compilations.  There may be a better way to do this?

Reply:
Ok, 451bc0f actually resolves this issue for both HCC / HIP-Clang and I have tested this works a system that doesn't have a GPU.

I opted for the "SET_SOURCE_FILES_PROPERTIES" path for both, as otherwise the unit tests seem to inherit the compiler options for `kokkos_gtest`, which breaks compilation of tests that have kernels.